paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/FlinkSinkBuilder.java

```diff
@@ -102,4 +104,13 @@ private DataStreamSink<?> buildForFixedBucket() {
                 new FileStoreSink(table, lockFactory, overwritePartition, logSinkFunction);
         return sink.sinkFrom(partitioned);
     }
+
+    private DataStreamSink<?> buildUnawareBucketSink() {
+        return new UnawareBucketWriteSink(
```

Comment:
Just inline `UnawareBucketWriteSink` here?

Reply:
add checks for append-only mode